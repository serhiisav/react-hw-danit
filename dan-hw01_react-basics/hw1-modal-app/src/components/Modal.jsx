import React, { Component } from "react";

export default class Modal extends Component {

    render() {
        const { isOpened, onClick, header, text, closeButton, actions, backgroundColor } = this.props;
        return (
            <div className={isOpened ? "modal-wrap active" : "modal-wrap"} onClick={onClick}>
                <div className="modal-body" onClick={(e) => e.stopPropagation()} style={{ backgroundColor: backgroundColor }}>
                    <div className="modal-header">
                        <p>{header}</p>

                        {closeButton && <span onClick={onClick} className="modal-close">âœ•</span>}
                    </div>
                    <p className="modal-text">{text}</p>
                    <div className="modal-buttons-wrap">{actions()} </div>
                </div>
            </div>
        )
    }
}