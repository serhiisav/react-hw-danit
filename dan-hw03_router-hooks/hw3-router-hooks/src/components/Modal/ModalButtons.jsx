import React from "react";

const ModalButtons = ({ dataModal, setModalStatus, handleClickModalOk }) => {
    const { btnActionOk, btnActionCancel } = dataModal;

    return (
        <>
            <button onClick={handleClickModalOk} id='btnModalOk' className='btnModal' type='button'>{btnActionOk}</button>
            <button onClick={setModalStatus} id='btnModalCancel' className='btnModal' type='button'>{btnActionCancel}</button>
        </>
    )
}

export default ModalButtons;