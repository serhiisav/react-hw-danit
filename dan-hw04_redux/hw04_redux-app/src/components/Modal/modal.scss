@import "../../variables";

.modal-wrap {
  height: 100vh;
  width: 100vw;
  background-color: #1d1d1de0;

  backdrop-filter: blur(5px);
  -webkit-backdrop-filter: blur(5px);

  position: fixed;
  top: 0;
  left: 0;
  display: none;
  z-index: 100;
}

.modal-wrap.active {
  display: flex;
  justify-content: center;
  align-items: center;
}

.modal-body {
  overflow-y: auto;
  max-width: 400px;
  max-height: 400px;
  border-radius: 10px;

  -ms-overflow-style: none;
  scrollbar-width: none;
  &::-webkit-scrollbar {
    display: none;
  }
}

.modal-header {
  color: $colorModalButton;
  border-radius: 10px;
  padding: 10px 20px;
  background-color: $bgModal;
  display: flex;
  justify-content: space-between;
  & > p {
    margin: 0;
  }
}
.modal-text {
  padding: 0 20px;
}
.modal-close {
  cursor: pointer;
}

.modal-buttons-wrap {
  display: flex;
  justify-content: center;
}

.btnModal {
  display: block;
  min-width: 100px;
  background-color: $bgModal;
  padding: 5px;
  margin: 10px;
  border-radius: 5px;
  border: none;
  color: $colorModalButton;
  font-size: medium;
  cursor: pointer;
  &:hover {
    opacity: 0.5;
    color: #000000;
  }
}

.btnModalRedirect {
  display: block;
  width: 100%;
  height: 100%;
}
