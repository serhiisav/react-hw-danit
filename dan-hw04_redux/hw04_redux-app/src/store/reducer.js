import { initialState } from "./store"

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case 'ADD_ITEMS_STARTED': {
            return { ...state, isLoading: true }
        }
        case 'ADD_ALL_ITEMS': {
            return { ...state, products: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_FAV_LIST': {
            return { ...state, itemsFavoriteList: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_CART_LIST': {
            return { ...state, itemsCartList: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_FAILURE': {
            return { ...state, isLoading: false, error: action.payload }
        }
        case 'SET_FAV_LIST': {
            return { ...state, favoritesList: action.payload }
        }
        case 'SET_CART_LIST': {
            return { ...state, cartList: action.payload }
        }
        case 'CHANGE_FAV_LIST': {
            return {
                ...state,
                favoritesList: state.favoritesList.includes(action.payload) ?
                    state.favoritesList.filter(el => el !== action.payload) :
                    [...state.favoritesList, action.payload]
            }
        }
        case 'CHANGE_CART_LIST': {
            return {
                ...state,
                cartList: state.cartList.includes(action.payload) ?
                    state.cartList.filter(el => el !== action.payload) :
                    [...state.cartList, action.payload]
            }
        }
        case 'SET_IS_OPENED_MODAL': {
            return { ...state, isOpenedModal: action.payload }
        }
        case 'SET_DATA_MODAL': {
            return { ...state, dataModal: action.payload }
        }
        default: {
            return state;
        }
    }
}

export default reducer;