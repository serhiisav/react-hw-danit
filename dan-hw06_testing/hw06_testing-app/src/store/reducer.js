
export const initialState = {
    products: [],
    favoritesList: [],
    cartList: [],
    isOpenedModal: false,
    isLoading: false,
    error: null,
    itemsFavoriteList: [],
    itemsCartList: [],
    isActiveOrderForm: false,
    orderInfo: {},
    totalsOrder: null,
    dataModal: {
        header: 'Modal',
        closeButton: true,
        text: 'Are you sure you want to delete it',
        btnActionOk: 'OK',
        backgroundColor: '#b46c7b',
        btnActionCancel: 'CANCEL',
    }
}

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case 'ADD_ITEMS_STARTED': {
            return { ...state, isLoading: true }
        }
        case 'ADD_ALL_ITEMS': {
            return { ...state, products: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_FAV_LIST': {
            return { ...state, itemsFavoriteList: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_CART_LIST': {
            return { ...state, itemsCartList: action.payload, isLoading: false, error: null }
        }
        case 'REMOVE_ITEMS_CART_LIST': {
            return { ...state, itemsCartList: [] }
        }
        case 'REMOVE_CART_LIST': {
            return { ...state, cartList: [] }
        }
        case 'ADD_ITEMS_FAILURE': {
            return { ...state, isLoading: false, error: action.payload }
        }
        case 'SET_FAV_LIST': {
            return { ...state, favoritesList: action.payload }
        }
        case 'SET_CART_LIST': {
            return { ...state, cartList: action.payload }
        }
        case 'CHANGE_FAV_LIST': {
            return {
                ...state,
                favoritesList: state.favoritesList.includes(action.payload) ?
                    state.favoritesList.filter(el => el !== action.payload) :
                    [...state.favoritesList, action.payload]
            }
        }
        case 'CHANGE_CART_LIST': {
            return {
                ...state,
                cartList: state.cartList.includes(action.payload) ?
                    state.cartList.filter(el => el !== action.payload) :
                    [...state.cartList, action.payload]
            }
        }
        case 'SET_IS_OPENED_MODAL': {
            return { ...state, isOpenedModal: action.payload }
        }
        case 'SET_DATA_MODAL': {
            return { ...state, dataModal: action.payload }
        }
        case 'SET_IS_ACTIVE_ORDER_FORM': {
            return { ...state, isActiveOrderForm: action.payload.isActive }
        }
        case 'SET_ORDER_INFO': {
            return { ...state, orderInfo: { ...action.payload.values, totalSum: state.totalsOrder, orderList: state.itemsCartList } }
        }
        case 'SET_TOTALS_ORDER': {
            return { ...state, totalsOrder: action.payload.orderSum }
        }
        default: {
            return state;
        }
    }
}

export default reducer;