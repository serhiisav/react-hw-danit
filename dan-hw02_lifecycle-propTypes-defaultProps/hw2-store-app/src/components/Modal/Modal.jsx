import React, { Component } from "react";
import './modal.scss';
import PropTypes from 'prop-types';


export default class Modal extends Component {

    render() {
        const { isOpened, setModalStatus, header, text, closeButton, actions, backgroundColor } = this.props;
        return (
            <div className={isOpened ? "modal-wrap active" : "modal-wrap"} onClick={setModalStatus}>
                <div className="modal-body" onClick={(e) => e.stopPropagation()} style={{ backgroundColor: backgroundColor }}>
                    <div className="modal-header">
                        <p>{header}</p>
                        {closeButton && <span onClick={setModalStatus} className="modal-close">âœ•</span>}
                    </div>
                    <p className="modal-text">{text}</p>
                    <div className="modal-buttons-wrap">{actions()} </div>
                </div>
            </div>
        )
    }
}

Modal.propTypes = {
    header: PropTypes.string,
    text: PropTypes.string,
    closeButton: PropTypes.bool,
    isOpened: PropTypes.bool,
    setModalStatus: PropTypes.func,
    actions: PropTypes.func,
    backgroundColor: PropTypes.string
}

Modal.defaultProps = {
    header: 'Modal',
    closeButton: true,
    text: 'Are you sure you want to delete it',
    btnActionOk: 'OK',
    backgroundColor: '#b46c7b',
    btnActionCancel: 'CANCEL',
}