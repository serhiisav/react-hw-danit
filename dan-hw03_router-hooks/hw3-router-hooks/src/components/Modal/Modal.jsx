import React from "react";
import './modal.scss';
import PropTypes from 'prop-types';
import ModalButtons from "./ModalButtons";


function Modal({ dataModal, isOpened, setModalStatus, handleClickModalOk }) {
    const { header, closeButton, text, backgroundColor } = dataModal;
    return (
        <div className={isOpened ? "modal-wrap active" : "modal-wrap"} onClick={setModalStatus}>
            <div className="modal-body" onClick={(e) => e.stopPropagation()} style={{ backgroundColor: backgroundColor }}>
                <div className="modal-header">
                    <p>{header}</p>
                    {closeButton && <span onClick={setModalStatus} className="modal-close">âœ•</span>}
                </div>
                <p className="modal-text">{text}</p>
                <div className="modal-buttons-wrap">{<ModalButtons handleClickModalOk={handleClickModalOk} setModalStatus={setModalStatus} dataModal={dataModal} />} </div>
            </div>
        </div>
    )
}


Modal.propTypes = {
    dataModal: PropTypes.object,
    isOpened: PropTypes.bool,
    setModalStatus: PropTypes.func,
    handleClickModalOk: PropTypes.func
}

Modal.defaultProps = {
    dataModal: {
        header: 'Modal',
        closeButton: true,
        text: 'Are you sure you want to delete it',
        btnActionOk: 'OK',
        backgroundColor: '#b46c7b',
        btnActionCancel: 'CANCEL',
    }
}

export default Modal;