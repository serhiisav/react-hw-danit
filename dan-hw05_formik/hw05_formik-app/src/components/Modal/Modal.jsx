import React from "react";
import './modal.scss';
import ModalButtons from "./ModalButtons";
import useOpenModal from "../../hooks/useOpenModal";
import { useSelector } from "react-redux";


function Modal() {
    const isOpenedModal = useSelector(state => state.isOpenedModal);
    const dataModal = useSelector(state => state.dataModal);

    const { setModalStatus } = useOpenModal();
    const { header, closeButton, text, backgroundColor } = dataModal;

    return (
        <div className={isOpenedModal ? "modal-wrap active" : "modal-wrap"} onClick={setModalStatus}>
            <div className="modal-body" onClick={(e) => e.stopPropagation()} style={{ backgroundColor: backgroundColor }}>
                <div className="modal-header">
                    <p>{header}</p>
                    {closeButton && <span onClick={setModalStatus} className="modal-close">âœ•</span>}
                </div>
                <p className="modal-text">{text}</p>
                <div className="modal-buttons-wrap">{
                    <ModalButtons />
                }
                </div>
            </div>
        </div>
    )
}

export default Modal;