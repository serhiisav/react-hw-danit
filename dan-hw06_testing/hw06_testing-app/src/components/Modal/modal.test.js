import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Modal from './Modal';
import reducer from '../../store/reducer';
import { createStore } from 'redux';

import '@testing-library/jest-dom/extend-expect';

describe('<Modal/> component', () => {

    it('should be displayed', () => {
        const initialState = {
            isOpenedModal: true,
            dataModal: {
                header: 'Modal',
                closeButton: true,
                text: 'Test text Modal',
                btnActionOk: 'OK',
                backgroundColor: '#b46c7b',
                btnActionCancel: 'CANCEL'
            }
        }
        const store = createStore(
            reducer,
            initialState,
        );

        const { container } = render(
            <Provider store={store}>
                <Modal />
            </Provider>);

        expect(container.firstChild).toHaveClass('modal-wrap active', { exact: true });
        expect(container.firstChild).toMatchSnapshot();
    })

    it('should not be displayed', () => {
        const initialState = {
            isOpenedModal: false,
            dataModal: {
                header: 'Modal',
                closeButton: true,
                text: 'Test text Modal',
                btnActionOk: 'OK',
                backgroundColor: '#b46c7b',
                btnActionCancel: 'CANCEL'
            }
        }
        const store = createStore(
            reducer,
            initialState,
        );

        const { container } = render(
            <Provider store={store}>
                <Modal />
            </Provider>
        );
        expect(container.firstChild).toHaveClass('modal-wrap', { exact: true });
        expect(container.firstChild).toMatchSnapshot();
    })

})