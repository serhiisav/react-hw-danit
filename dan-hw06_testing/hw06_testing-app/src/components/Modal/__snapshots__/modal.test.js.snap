// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> component should be displayed 1`] = `
<div
  class="modal-wrap active"
>
  <div
    class="modal-body"
    style="background-color: rgb(180, 108, 123);"
  >
    <div
      class="modal-header"
    >
      <p>
        Modal
      </p>
      <span
        class="modal-close"
      >
        ✕
      </span>
    </div>
    <p
      class="modal-text"
    >
      Test text Modal
    </p>
    <div
      class="modal-buttons-wrap"
    >
      <button
        class="btnModal"
        id="btnModalOk"
        type="button"
      >
        OK
      </button>
      <button
        class="btnModal"
        id="btnModalCancel"
        type="button"
      >
        CANCEL
      </button>
    </div>
  </div>
</div>
`;

exports[`<Modal/> component should not be displayed 1`] = `
<div
  class="modal-wrap"
>
  <div
    class="modal-body"
    style="background-color: rgb(180, 108, 123);"
  >
    <div
      class="modal-header"
    >
      <p>
        Modal
      </p>
      <span
        class="modal-close"
      >
        ✕
      </span>
    </div>
    <p
      class="modal-text"
    >
      Test text Modal
    </p>
    <div
      class="modal-buttons-wrap"
    >
      <button
        class="btnModal"
        id="btnModalOk"
        type="button"
      >
        OK
      </button>
      <button
        class="btnModal"
        id="btnModalCancel"
        type="button"
      >
        CANCEL
      </button>
    </div>
  </div>
</div>
`;
